internal/services/pam/pam_test.go

```diff
@@ -815,40 +815,53 @@ func startSession(t *testing.T, client authd.PAMClient, username string) string
 	return sbResp.GetSessionId()
 }
 
-func TestMain(m *testing.M) {
-	if os.Getenv("GO_WANT_HELPER_PROCESS") != "" {
-		os.Exit(m.Run())
-	}
-
-	slog.SetLogLoggerLevel(slog.LevelDebug)
-
+// setupGlobalBrokerMock creates and points to a test-wide system bus, registering the mock broker on it.
+func setupGlobalBrokerMock() (cleanup func(), err error) {
 	// Start system bus mock.
 	busCleanup, err := testutils.StartSystemBusMock()
 	if err != nil {
-		fmt.Fprintf(os.Stderr, "%v\n", err)
-		os.Exit(1)
+		return cleanup, err
 	}
-	defer busCleanup()
+	cleanup = busCleanup
 
 	// Start brokers mock over dbus.
-	brokersConfPath, cleanup, err := initBrokers()
+	brokersConfPath, brokerCleanup, err := initBrokers()
 	if err != nil {
-		fmt.Fprintf(os.Stderr, "%v\n", err)
-		os.Exit(1)
+		return cleanup, err
```

Comment:
Isn't better to return broker cleanup func anyways?

Reply:
This was my first approach, but I think if we change the function later, it will be harder to miss this as we always return cleanup and grow it.